Homework3/Task2/Extencion Methods/Extension.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Task2.Classes;
+using Task2.Interfaces;
+
+namespace Task2.ExtencionMethods
+{
+    public static class Extension
+    {
+        public static Stack<T> Reverse<T>(this IStack<T> stack)
+        {
+            Stack<T> newStack = new Stack<T>();
```

Comment:
simplify the method, do not case to Stack. 
It is OK that original "stack" object will be empty after the Reverse execution, 

Use IsEmpty method, it allows you to ensure that original stack still has elements. 
Do not access internal field "stack", just call Pop.